### ISSUE TITLE
Glx version get segfault

### ISSUE THREAD

**Issue by NotAFile** [xtreme8000/aof-opengl#60]
You know the one with glx_major_version, where the workaround is `return 1`

**Comment by ghost** [comment]
@NotAFile could you describe in more details please?

**Comment by NotAFile** [comment]
sorry, this is so vague because @xtreme8000 is already aware of the issue.

Basically, due to some kind of race condition(?) the `glx_major_ver` implementation causes a segfault on linux.

the workaround is to replace it with `return 1`

**Comment by ghost** [comment]
@NotAFile @xtreme8000 Hmm.. That's interesting because when I built and ran the `client` everything was fine until I tried joining a server.
After the server connection had been established, the game window has crashed and I've noticed in the terminal window the welcome message from the server and the segmentation fault error.
Am I facing the same problem you are talking about?

**Comment by godwhoa** [comment]
@omh5  Yep
```
17:31:06 INFO  network.c:279: map data was 616318 bytes
17:31:07 INFO  network.c:298: cache/2474EE97.vxl
17:31:07 INFO  main.c:63: Welcome to Cikila
fish: “./client” terminated by signal SIGSEGV (Address boundary error)
```
Patch that works for me:
```patch
diff --git a/src/glx.c b/src/glx.c
index b3ec6d9..9ad7fc0 100644
--- a/src/glx.c
+++ b/src/glx.c
@@ -45,6 +45,7 @@ PFNGLGETUNIFORMLOCATIONPROC glGetUniformLocation;
 #endif
 
 static int glx_major_ver() {
+       return 1;
        #ifdef OPENGL_ES
        return 2;
        #else
```

**Comment by ghost** [comment]
@godwhoa thanks, I'll test it to see if it works for me.

**Comment by omh5** [comment]
@godwhoa could you please drop me a message on my email?

**Comment by ghost** [comment]
@godwhoa I have built the sources, but this time I used the workaround in the code. It worked! :smile:

**Comment by xtreme8000** [comment]
@omh5 great

**Comment by xtreme8000** [comment]
Closed by 3ef89a59eaf80d9df711f07957e2179b28996333
